src/MongoDB.Driver.Core/WriteConcern.cs

```diff
@@ -135,6 +135,13 @@ public static WriteConcern FromBsonDocument(BsonDocument document)
                 writeConcern = writeConcern.With(journal: j.ToBoolean());
             }
 
+            // seems hacky - addresses the specific new test added
+            if (w!=null && j!=null && w.AsInt32 == 0 && j.AsBoolean)
```

Comment:
I did this to pass the new test case that specifies `{"w": 0, "journal": true}` is invalid, but there may be other invalid parameter combinations that I am not aware of.

Reply:
Sorry, yet another area where I'll need to update the style guide :)  as with `+`, we prefer the `!=` operator to be wrapped with a single space e.g. `a != b`. Off the top of my head, this holds true for all the infix operators.

As I mentioned in the previous comment, I think this is the exact case we need to guard against, but let's open it up to team review and see what the rest of the team thinks! :)

Reply:
Sorry, yet another area where I'll need to update the style guide :) as with +, we prefer the != operator to be wrapped with a single space e.g. a != b. Off the top of my head, this holds true for all the infix operators.

As I mentioned in the previous comment, I think this is the exact case we need to guard against, but let's open it up to team review and see what the rest of the team thinks! :)

Reply:
Resolving this conversation since the whitespace was fixed and the team seems to agree that the case you specified is the case we need to guard against!